3IPs/3ip-2.md

```diff
@@ -9,92 +9,68 @@ created: 2018-11-20
 
-## Encryption
-
-We simply use `s` to symmetrically encrypt the entries in the thread if it's a private thread.
-
-## Thread subscriptions
-
-Users can subscribe to threads. This means that they will remember the thread next time they go online, and they can ask the network for updates. When a user subscribes to a thread it simply adds the thread-id and name to the rootStore of the users 3Box. When the user unsubscribes the thread entry is removed from the rootStore.
-
-## Threads in spaces?
-
-Is this something we want to enable? Maybe as a second iteration, it adds a lot of complexity.
+## Thread creation
+Threads are created inside of [Spaces](./3ip-1.md). A thread has a `name` which could be any string. For example the name could be the address of a contract if you want to create a thread for commenting on that contract. This also means that two different spaces could have their own threads with the same name. When a thread is created it gets the name `3box.thread.<space-name>.<thread-name>` for the orbitdb logstore.
 
 ## API
 
-**List threads the user is subscribed to:**
+All of the calls below assume that you have opened a space:
 ```js
-const threadNames = box.threads.list()
-console.log(threadNames)
-> ["3box-chat", "Bounties-issue432"]
+const space = box.openSpace('aGoodSpace')
 ```
 
-**Create a new thread:**
-
-The `open` parameter is a boolean that specifies if there is write permissions to the thread or not. Open threads anyone can post to, but there is no spam protection. Note that new threads are not automatically subscribed to.
+**List threads the user has interacted with**
+Lists all threads that the user has interacted with. Returns a list of objects that contains the name of the thread and if the thread is currently open (if the orbitdb store for it is open, and we are actively listening for updates).
 
 ```js
-const thread = await box.threads.new(<name>, <open=false>)
+const threadNames = space.threads.list()
+console.log(threadNames)
+> [{ name: "3box-chat", open: true }, { name: "Bounties-issue432", open: false }]
 ```
 
-**Get thread id**
-
-the thread-id is simply the orbitdb-address (String) of the underlaying DB.
+**Create an existing thread:**
+When a thread that hasn't been opened before is opened its orbitdb address is also added to the users space under the key `thread-<thread-name>`.
 
 ```js
-const threadId = thread.getId() 
+const thread = await threads.threads.open(<name>)
 ```
 
-**Subscribe to an existing thread:**
+**Create a new thread:**
+When a new thread is created its orbitdb address is also added to the users space under the key `thread-<thread-name>`.
 
 ```js
-const thread = await box.threads.subscribe(<thread-id>, <name>) 
+const thread = await threads.threads.new(<name>)
```

Comment:
is this weird to have threads.threads  ?

Reply:
oh, should be `space` of course :)